reference/vane-apis/behn.md

```diff
@@ -0,0 +1,366 @@
+    ::  behn must get activated before other vanes in a %wake
+    ::
+    ::    TODO: uncomment this case after switching %crud tags
+    ::
+    ::    We don't know how to handle other errors, so relay them to %dill
+    ::    to be printed and don't treat them as timer failures.
+    ::
+    ::  ?.  =(%wake tag)
+    ::    ~&  %behn-crud-not-first-activation^tag
+    ::    [[duct %slip %d %flog %crud tag error]~ state]
+    ::
+    ?:  =(~ timers.state)  ~|  %behn-crud-no-timer^tag^error  !!
+    ::
+    (wake `error)
+ ```
+ 
+
+### `%drip`
+
+`%drip` is utilized to delay `%give`ing a `gift`.
+
+`%drip` allows one to delay `gift`s until a given condition is met. For example,
+if Clay crashes and you do not wish for an app to then crash when it tries to
+access Clay before it is restarted, `%drip` can delay that access until Clay has
+been restored. 
+
+`%drip` only handles `gift`s, and can only schedule `gift`s for as soon as
+possible after the prescribed condition is met.
+
+#### Accepts
+
+```hoon
+mov=vase
+```
+
+`%drip` takes in a `%give` `move` in a `vase`.
+
+#### Returns
+
+In response to a `%drip` `task`, Behn will `%give` a `%meta` `gift` containing
+the `gift` originally `%give`n to Behn when `%drip` was first called.
+
+Is that really correct? It looks like the gift returned is actually a response to
+a pass it passed itself, not the original event?
+
+#### Source
+
+```hoon
+  ::  +drip:  XX
+  ::
+  ++  drip
+    |=  mov=vase
+    =<  [moves state]
+    ^+  event-core
+    =.  moves
+      [duct %pass /drip/(scot %ud count.drips.state) %b %wait +(now)]~
+    =.  movs.drips.state
+      (~(put by movs.drips.state) count.drips.state mov)
+    =.  count.drips.state  +(count.drips.state)
+    event-core
+```
+
+
+
+### `%huck`
+
+Immediately gives back an input move.
+
+#### Accepts
+
+```hoon
+mov=vase
+```
+
+Behn takes in a `move` contained in a `vase`.
+
+#### Returns
+
+```hoon
+[%give %meta mov]
+```
+
+Behn returns the input `move` as a `%meta` `gift`.
+
+#### Source
+
+```hoon
+  ::  +huck: give back immediately
+  ::
+  ::    Useful if you want to continue working after other moves finish.
+  ::
+  ++  huck
+    |=  mov=vase
+    =<  [moves state]
+    event-core(moves [duct %give %meta mov]~)
+```
+
+
+
+### `%rest`
+
+Cancels a timer that was previously set.
+
+Behn takes in a `@da` and cancels the timer at that time if it exists, then
+adjusts the next wakeup call from Unix if necessary.
+
+#### Task
+
+```hoon
+@da
+```
+
+
+#### Gift
+
+This `task` returns no `gift`s.
+
+
+#### Source
+
+```hoon
+::  +rest: cancel the timer at :date, then adjust unix wakeup
+++  rest  |=(date=@da set-unix-wake(timers.state (unset-timer [date duct])))
+```
+
+
+
+### `%trim`
+
+This `task` is sent by the interpreter in order to free up memory.
+ This `task` is empty for Behn, since it is not a good idea to forget your timers.
+
+#### Accepts
+
+This `task` has no arguments. 
+
+#### Returns
+
+This `task` returns `[moves state]`.
```

Comment:
`[moves state]` without adding anything to `moves` is just a no-op.  It doesn't return anything.